### ISSUE TITLE
(discussion) Statement about community involvement appreciated

### ISSUE THREAD

**Issue by choeller** [angular/angular#8615]
Looking at the commit history of the master over the last couple of month there are almost no commits not created by a core member of the team.(not a single one in the last month). In contrast to that there are dozens of open pull-request dealing with really anoying bugs, that are just not taken care of by anyone. On most of them no one even reacted in any way. 

To be honest this is super *frustrating and discouraging*. I personally openend a pull request several weeks ago fixing async validation for template driven forms and got no feedback at all(https://github.com/angular/angular/pull/8186), and there are others that have put way more energy into fixing bugs that are just not worshiped. (https://github.com/angular/angular/pull/7152, https://github.com/angular/angular/pull/7580 etc.) Even worse most of them are now in conflict state because of the major refactoring for RC, so the work people invested into creating bugfixes was simply ignored before changing the whole project structure... 

I would really appriciate some statement by the core team on how community involvement is planned to be managed. The way it's currently handled IMO is really sad for a project stating on their website: 

> We'd love for you to contribute to our source code and to make Angular projects even better.

That being said, looking at other projects like react the way of communicating with the community is way more encouraging - all pull-request are labeled, and core team-members *do* answer on bugs and questions. I'm really curious if anyone will respond to this message, but I definetly think I'm not the only one being unsatisfied with the current state of community involvement...

**Comment by fxck** [comment]
I'm just gonna leave this here, since it's partially related. I had a discussion with @pkozlowski-opensource couple of months ago, back then I was quite frustrated with the progress being done on router, and a lack of any information about what's happening(and it proved to be very valid concern). You can read it here, around this message https://gitter.im/angular/angular?at=5674120b37169e856a757906. 

I keep hearing on conferences how much the team listens to the community, takes their feedback etc. and I wonder, what community? Internal google teams maybe? Being fairly frequent user of one of the biggest communities, that is gitter, I don't think I've seen any team members or community managers being involved whatsoever. Except for @robwormald and @alxhub of course.

**Comment by kylecordes** [comment]
I think the conference talks should be thought of as primarily aspirational and inspirational. If you go read the actual contributing guidelines:

https://github.com/angular/angular/blob/master/CONTRIBUTING.md

They are well written and reasonably clear. For example, when you get to the part about contributing code, you see:

"If you would like to implement a new feature, please submit an issue with a proposal for your work first, to be sure that we can use it. Angular 2 is in developer preview and we are not ready to accept major contributions ahead of the full release. Please consider what kind of change it is:"

I interpret this as saying that after they get the full release, there will be more focus on accepting code changes from outside the core team.

Also keep in mind, Google has quite a large team on the project, and just coordinating that group's work to get a steady succession of forward overall progress, is not trivial.

**Comment by choeller** [comment]
@kylecordes I can only speak for myself, but the PR I created is directly linked to a created issue/bug. (https://github.com/angular/angular/issues/8022) It's a single-line fix and in the issue I explicetly asked the pariticapting core members if they could have a look at this, so this would have been a task of 30 seconds... 

> I interpret this as saying that after they get the full release, there will be more focus on accepting code changes from outside the core team.

If this is the case they should say it like this - it's clear that they won't accept something like "hey I rewrote the animation core" but most of the open PRs are dealing with serious and isolated bugs and would just improve the framework... 

> Also keep in mind, Google has quite a large team on the project, and just coordinating that group's work to get a steady succession of forward overall progress, is not trivial.

IMO the fact that they have a "large team on the project" is an argument for the exact opposite - if one single person of this  team would take 30 minutes a day checking github and responding to issues there would not be the impression of total ignorance in the community

**Comment by PascalPrecht** [comment]
I understand your frustration and we can totally do better here. We just shouldn't forget that the team has to build and maintain over 2000 apps *while* developing A2, which of course causes the project to be mainly driven by the needs on these apps.

I'll forward this to Naomi and see what we can do.

**Comment by choeller** [comment]
@PascalPrecht I'm of course completely aware that there are a number of *other* stackholders than the community involved, but IMO this is no excuse for not caring about this part at all. On the long run this community will - or will not - be the main group of people using the framework. 
I personally already invested hundreds of hours in learning / playing around with the framework / answering questions on gitter / creating issues and I lately really thought about just stopping that, wait for a couple of month and see if it's worth coming back then. I decided against that because that's just not the way open-source works... 

So talking about this one thing about your answer kind of scares me:

>  which of course causes the project to be mainly driven by the needs on these apps

This is honestly the complete opposite of what is promoted on conferences like ng-conf.

Don't get me wrong - I really appreciate that you are trying to rise some awareness on this, it's just that I more and more get the feeling things are heading into the wrong direction...

**Comment by PascalPrecht** [comment]
> This is honestly the complete opposite of what is promoted on conferences like ng-conf.

I'm sorry, maybe I didn't make myself very clear. What I was trying to say is that prolly most of the time goes into building things currently, which is why there's not enough left to check community efforts at this time. I forwarded your feedback, hold your horses :)

**Comment by choeller** [comment]
I really hate to sound like a nit-picker here, but it's just important for me that this thing here is not leading to an - this guy is just writing down his frustration  - impression. My main point is that the community does not feel like being taken serious, so "holding my horses" would have meant to just not open this ticket and accept everything as is... I'm still looking forward to hear from the core-members and see if it's possible to improve the current situation...

**Comment by robwormald** [comment]
@choeller this is good timing, as we had a long meeting yesterday about process improvements for this kind of thing. I'm going to write up a post on where we're at and where we're going, and this will certainly be covered.

tldr - we definitely hear you, thanks for the support, and we *will* sort it out.

**Comment by mhevery** [comment]
@choeller I am sorry that you are upset and that your PR is not getting the attention it deserves. The short answer is that we are overwhelmed with all of the things which need to be done. Yes, I understand that your issue seems deer to you, but please understand that there are higher priority issues which the core team is occupied with.

Yes, we have dropped the ball on the PRs and Issues coming in. And I understand why you may think that it would take 30 seconds, but nothing takes 30 seconds. When you make statements like that you hi-jack my brain to think how unrealistic 30 seconds is rather then spending time helping the situation.

Here is what needs to happen:
1. I reviewed the #8186 and determined that I don't understand why you made those changes, and that you need to have an unit tests rather then e2e test. A proper test should explain what situation you are trying to avoid, rather then that x should work. 
2. Once we have a PR which we approve it goes to caretaker.
3. The caretaker's job is to get the change into Google and make sure it does not break any projects (we have many major internal customers which we must support). This takes about 10 mins of work and about 2 hrs of baby sitting. We usually batch several small changes together which we consider low risk, such as the one you propose.
4. Once google projects are happy the PR get's merged into master, and PR is closed.


As you can see this is not 30 seconds worth of work. Also we have been doing major refactorings to get ready for RC, so we have been trying to minimize unnecessary changes.

**Comment by choeller** [comment]
@mhevery @robwormald thanks for the head up... 

I have to admit that this "30 seconds" thing was kind of a stupid thing to say, and I totally see that there is more work involved. ;) Also this one pull-request is really not my major concern - it's rather that looking at the list of issues and PRs I more and more get the impression that the energy the community puts into participating is kind of meaningless. 

Having said that It's good to hear you are working on sorting these things out and I'm curious on how things will evolve...

**Comment by fdelayen** [comment]
@mhevery Thank you for taking time to explain that process !

**Comment by pkozlowski-opensource** [comment]
I think that we caught up with the PR queue. Issues are more challenging but all the community help with those is more than appreciated!
